src/features/hotspots/settings/HotspotDiagnosticsConnection.tsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import {
+  ActivityIndicator,
+  LayoutAnimation,
+  Linking,
+  Platform,
+} from 'react-native'
+import { Device } from 'react-native-ble-plx'
+import { useSelector } from 'react-redux'
+import Text from '../../../components/Text'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import Box from '../../../components/Box'
+import { useConnectedHotspotContext } from '../../../providers/ConnectedHotspotProvider'
+import HotspotItem from './HotspotItem'
+import { RootState } from '../../../store/rootReducer'
+import { useBluetoothContext } from '../../../providers/BluetoothProvider'
+import useAlert from '../../../utils/useAlert'
+import sleep from '../../../utils/sleep'
+
+type Props = { onConnected: (hotspot: Device) => void }
+const HotspotDiagnosticsConnection = ({ onConnected }: Props) => {
+  const [scanComplete, setScanComplete] = useState(false)
+  const [enabled, setEnabled] = useState(false)
+  const [selectedHotspot, setSelectedHotspot] = useState<Device | undefined>()
+  const {
+    scanForHotspots,
+    availableHotspots,
+    connectAndConfigHotspot,
+  } = useConnectedHotspotContext()
+  const { enable, getState } = useBluetoothContext()
+  const { showOKCancelAlert } = useAlert()
+
+  const hotspotCount = Object.keys(availableHotspots).length
+  const keys = Object.keys(availableHotspots)
+  const { t } = useTranslation()
+
+  const { connectedHotspot } = useSelector((state: RootState) => state)
+
+  useEffect(() => {
+    const handleConnect = async () => {
+      if (selectedHotspot && !!connectedHotspot.address) {
+        await sleep(500)
```

Comment:
Would there be any difference here is doing a setTimeout() instead of a sleep? Did this come from the old code?

Reply:
I think it would have the same effect. `sleep` is a function I added. It's just a wrapper around setTimeout.

`export default (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))`

Reply:
Ah cool, looks good.